### PR TITLE
Clarify WebGL 2 spec for attribute aliasing

### PR DESCRIPTION
**Pull request by Oletus**
GLSL ES 3.00.6 already has strict rules when it comes to attribute
aliasing, so WebGL 2.0 can simply refer to those rather than spell out
its own.

A new test is added to fully cover the attribute aliasing rules of
GLSL ES 3.00.6. It shares some code with the older attribute aliasing
test for WebGL 1.0.

### CODE REVIEW

**Comment by Oletus** [comment]
This is one possible way to resolve conflicts between attribute aliasing rules in GLSL ES 3.00.6 vs WebGL 2.0.

On the other hand, if we want to ensure better WebGL backwards compatibility, we could opt for alternative, relaxed rules for WebGL 2.0 that would be more in line with WebGL 1.0 rules. This would also make the implementation simpler, since the set of attributes to validate would not be dependent on shader version. Any thoughts?

**Comment by zhenyao** [comment]
I like the alternative where we allow aliasing but enforcing the WebGL 1.0 rule, a link error must occur when more than one of the aliased attributes is active in the executable program.

**Comment by Oletus** [comment]
Here's an alternative pull request making the WebGL 2.0 rules consistent with WebGL 1.0 instead: https://github.com/KhronosGroup/WebGL/pull/2619

**Comment by Oletus** [comment]
(also fixed missing JS file in this patch, in case we after all decide to go with this alternative)

**Comment by zhenyao** [comment]
LGTM, but let's wait to hear others' opinions on this.

**Comment by Oletus** [comment]
@zhenyao To be clear - do you mean that the other alternative looks good? Maybe further discussion should be on the other pull request.

**Comment by zhenyao** [comment]
Yes, I meant the alternative. Sorry about the confusion.

**Comment by Oletus** [comment]
So, are we going forward with this alternative as discussed last Thursday? @jdashg @kenrussell ?

File: specs/latest/2.0/index.html [hunk 1]
```diff
@@ -3527,13 +3527,10 @@ <h3><a name="CLIENT_WAIT_SYNC">clientWaitSync</a></h3>
...
     <h3><a name="VERTEX_ATTRIBUTE_ALIASING">Vertex Attribute Aliasing</a></h3>
     <p>
-        In the WebGL 2.0 API, if more than one active attribute name is bound to the same location, it is considered
-        aliased, regardless whether there exists a path through the shader that consumes more than of these
-        attributes. Such definition of aliasing is broader than what is defined in
-        <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_spec_3.0.4.pdf#nameddest=section-2.11.5">OpenGL ES 3.0.4 &sect;2.11.5</a>)</span>.
-    </p>
-    <p>
-        A link error is required when aliasing exists.
+        WebGL 2.0 API implementations must strictly follow GLSL ES 3.00.6 section 12.46, that
```
Comment (jdashg) [inline specs/latest/2.0/index.html]: s/that/which/

File: specs/latest/2.0/index.html [hunk 2]
```diff
@@ -3527,13 +3527,10 @@ <h3><a name="CLIENT_WAIT_SYNC">clientWaitSync</a></h3>
...
+        specifies that any vertex attribute aliasing is disallowed. In addition, WebGL 1.0 spec
+        section <a href="../1.0/index.html#ATTRIBUTE_ALIASING">Attribute aliasing</a> applies to
+        GLSL ES 1.00 shaders used with the WebGL 2.0 API.
```
Comment (jdashg) [inline specs/latest/2.0/index.html]: "As stated in ES 3.0.5 p57, ESSL1 shaders may still alias, as allowed by the WebGL 1.0 spec."

**Comment by Oletus** [comment]
Addressed comments, merging this.

**Comment by kenrussell** [comment]
LGTM after the fact. The expanded test looks good. Sorry for the delay reviewing.

**Pull request merged**